pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript.md

```diff
@@ -3,120 +3,141 @@ $title: Use custom JavaScript in AMP pages
 
-The [AMP cache](../../../documentation/guides-and-tutorials/learn/amp-caches-and-cors/how_amp_pages_are_cached.md) serves custom JavaScript files inserted with `<amp-script>` the same way it serves AMP component scripts, which ensures that any custom JavaScript will not slow the speed of an AMP document.
+`amp-script` is essentially a wrapper around WorkerDOM that makes WorkerDOM usable in AMP. WorkerDOM provides the core of `amp-script`'s functionality.
 
-The AMP cache proxies the JavaScript files and then delivers them. Users can expect the same performance experience from a page using `<amp-script>` as a page that doesn't include it.
 
-# Using `<amp-script>`
+## Capabilities
 
-To guarantee AMP pages will load consistently fast and with smooth UIs, limitations exist on `<amp-script>`.
+The JavaScript language is the same in a worker as it is elsewhere in the browser. Thus, in `amp-script`, you can use JavaScript in the way that you're used to. WorkerDOM also recreates many commonly used DOM APIs and makes them available for your use. It supports common Web APIs like `Fetch` and `Canvas`, and it gives you to selected global objects like `navigator` and `localStorage`. You can assign handlers for browser events in the usual way.
 
-## Initialization
+However, `amp-script` does not support the entire DOM API or Web API, as this would make its own JavaScript too large and cumbersome. See [the documentation](../../../documentation/components/reference/amp-script.md#supported-apis) for details, and refer to [these samples](https://amp.dev/documentation/examples/components/amp-script/) to see `amp-script` in use.
 
-JavaScript inside the Web Worker allows minimal change to the DOM on load. Changes allowed during this phase are:
+A handful of synchronous DOM API methods are replaced with alternatives that return a Promise. For example,  `getBoundingClientRect()` is replaced by `getBoundingClientRectAsync()`.
 
-*   Registering event handlers.
-*   Splitting a TextNode into multiple TextNodes, to allow for frameworks that require it.
+To maintain AMP's guarantees of performance and layout stability, `amp-script` comes with some restrictions. Your code is discouraged from making potentially distracting mutations that do not follow a user interaction. You can't add stylesheets or additional scripts to the DOM, and [`importScripts()`](https://developer.mozilla.org/en-US/docs/Web/API/WorkerGlobalScope/importScripts) is not supported. See [the documentation](../../../documentation/components/reference/amp-script.md#user-gestures) for details.
```

Comment:
"Your code is discouraged from making potentially distracting mutations that do not follow a user interaction." This sentence is a little wishy-washy with "discouraged" and "potentially" and is also passive. Please rephrase.

Reply:
"cannot make mutations" would be valid